File: frontend/services/health.ts
import { apiClient } from "../lib/api-client";

export interface GeneralHealthResponse {
  status: string;
  service: string;
  environment: string;
  version: string;
}

export interface DatabaseHealthResponse {
  status: string;
}

/**
 * Fetches the general backend health status.
 */
export async function getGeneralHealth(): Promise<GeneralHealthResponse> {
  const response = await apiClient.get<GeneralHealthResponse>("/health");
  return response.data;
}

/**
 * Fetches the backend's database connection status.
 */
export async function getDatabaseHealth(): Promise<DatabaseHealthResponse> {
  const response = await apiClient.get<DatabaseHealthResponse>("/health/database");
  return response.data;
}

File: backend/schemas/health.py
from pydantic import BaseModel, Field

class HealthResponse(BaseModel):
    """
    Pydantic schema representing the general health check response.
    """
    status: str = Field(..., json_schema_extra={"example": "healthy"})
    service: str = Field(..., json_schema_extra={"example": "Nexora AI Support Hub API"})
    environment: str = Field(..., json_schema_extra={"example": "development"})
    version: str = Field(..., json_schema_extra={"example": "0.1.0"})

class DatabaseHealthResponse(BaseModel):
    """
    Pydantic schema representing the database health check response.
    """
    status: str = Field(
        ..., 
        json_schema_extra={"example": "connected"}, 
        description="Current MongoDB status: 'connected', 'unavailable', or 'not_configured'"
    )
